### PR TITLE
Add not exist greater than/ less then for segment filters

### PR DESCRIPTION
**Pull request by kuzmany**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Added segment filter not exist greater than/less then.
As follow up of this PR https://github.com/mautic/mautic/pull/7122/ (most recently https://github.com/mautic/mautic/pull/8652)

This allow use filters for example:

- contacts with orders but not last_active from last month etc

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2.  Create custom datetime 
3. If today is 14. 1. 2019 set this custom field to 10. 1. 2019
4. Got to segments with operator  `not exists greater than` 11 .1. 2019
5: Wait
6. Contact should be added to segment
7. Change the filter to not exists greater than 9. 1. 2019
8. Contact should be removed from segment
9. Repeat same for `not exist less than` operator

### CODE REVIEW

File: app/bundles/LeadBundle/Entity/OperatorListTrait.php [hunk 1]
```diff
@@ -88,10 +88,20 @@ trait OperatorListTrait
...
 
         return (null === $operator) ? $operatorOption : $operatorOption[$operator];
     }
 
+    /**
+     * Method for modify operators in each model separately.
+     *
+     * @param $operatorOption
+     */
+    public function modifyOperators($operatorOption)
```
Comment (alanhartless) [inline app/bundles/LeadBundle/Entity/OperatorListTrait.php]: I'm not sure I'm a fan of this. I know we use inherited trait methods in other places in older code but it's not considered very "clean" by today's standards, especially with a pass by referenced array. I think the cleaner solution for this scenario is to pass an optional array into getFilterExpressionFunctions that gets merged into `$operatorOption`.

**Review by alanhartless — CHANGES_REQUESTED** [review]

File: app/bundles/LeadBundle/Entity/OperatorListTrait.php (hunk 1 above)
Comment (alanhartless) [reply app/bundles/LeadBundle/Entity/OperatorListTrait.php]: Or consider refactoring away from this trait.

**Comment by npracht** [comment]
Hi there!
It has been decided to not create any extra Mautic 2 minor versions (which means no more features in M2) knowing that Mautic 3 is coming very soon.

**We now want to integrate your contribution in the Mautic 3 roadmap as 3.1.0 candidate.**

## How to do?
1. Check if your feature or enhancement is still missing and relevant in Mautic 3
2. **If Yes**: rebase your PR against the 3.x branch
3. **If No**: please consider closing your PR if you are absolutely sure that the feature has made it into Mautic 3 already

**Please report results by commenting on your PR to make us administration easier.**
______
You can more information on how to do all of that on [this blog post "Getting you PR ready for Mautic 3"](https://www.mautic.org/blog/developer/getting-your-pr-ready-mautic-3).

**Comment by RCheesley** [comment]
@cla-bot check

**Comment by kuzmany** [comment]
This PR need rewrite with use new listeners once after merge Segments refactoring https://github.com/mautic/mautic/pull/8649

Then I close it at the moment

**Pull request closed without merging**